solutions/ac-GCDOnBlackboard.mdx

```diff
@@ -0,0 +1,67 @@
+---
+id: ac-GCDOnBlackboard
+source: AC
+title: GCD On Blackboard
+author: Andrew Cheng
+---
+
+**Time Complexity:**
+$\mathcal{O}(N \log N)$ (GCD operations take $\mathcal{O}(\log N)$ time)
+
+## Solution
+The problem asks us to find the maximum possible GCD of the remaining $N-1$ numbers
+after taking any one of them away. Naively trying every single combination will result in a
+complexity of $\mathcal{O}(N^2)$ as there are $N$ different ways to take away a number and calculating
+the GCD will take $\mathcal{O}(N)$ time per case. 
```

Comment:
Why do you have O(N) here but O(N log N) above?

Actually, shouldn't the number of modulo operations be `O(N+\log(max a_i))`?

Reply:
My bad. Though shouldn't the time per case be ```O(N \log(max a_i))``` instead of ```O(N+\log(max a_i)) using brute force.

Reply:
There are only `log(max a_i)` distinct prefix GCDs though

Reply:
Uhh I don't quite understand why are there only ```log(max a_i)``` distinct prefix GCDs. I was thinking that the basic method would be to try out all N different numbers to take away and for each case one would have to calculate the GCD of the current case by scanning across the rest N-1 numbers for a total complexity of ```O(N \log(max a_i))``` per case.

Reply:
figure it out :)